### PR TITLE
Remove base

### PR DESCRIPTION
**Pull request by ncdc**
Remove base and only do the initialization logic necessary to each type
of upgrade.

Most of this is shuffling code around.

### CODE REVIEW

**Comment by ncdc** [comment]
FYI @rudoi

File: main.go [hunk 1]
```diff
@@ -137,20 +133,35 @@ type upgrader interface {
...
 	var (
 		log      = newLogger()
 		upgrader upgrader
 		err      error
 	)
 
-	switch config.TargetCluster.UpgradeScope {
-	case upgrade.ControlPlaneScope:
+	validUpgradeScope := false
+	validScopes := []string{controlPlaneScope, machineDeploymentScope}
```
[the remark this answers is not in the document]
Comment (ncdc) [reply main.go]: Oh let me fix it. I was copying code from the ValidateArgs function.

**Comment by ncdc** [comment]
This is definitely lower priority than fixing the configmap issue. I can rebase if needed.

File: main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ncdc) [reply main.go]: Updated

[the remark this answers is not in the document]
Comment (chuckha) [reply main.go]: ah got it, thanks!

File: pkg/upgrade/control_plane.go [hunk 2]
```diff
@@ -11,23 +11,26 @@ import (
 	"strings"
 	"time"
 
-	"sigs.k8s.io/yaml"
-
 	"github.com/blang/semver"
 	"github.com/go-logr/logr"
 	"github.com/pkg/errors"
-	"github.com/vmware/cluster-api-upgrade-tool/pkg/internal/kubernetes"
+	kubernetes2 "github.com/vmware/cluster-api-upgrade-tool/pkg/internal/kubernetes"
```
[the remark this answers is not in the document]
Comment (ncdc) [reply pkg/upgrade/control_plane.go]: Yeah, I don't have a better name (yet), and it was already in the code base

[the remark this answers is not in the document]
Comment (chuckha) [reply pkg/upgrade/control_plane.go]: i take blame for this. goland auto import naming >.>

**Comment by ncdc** [comment]
Added some commits (to be squashed before merging).

**Comment by ncdc** [comment]
Squashed & probably ready to go

**Pull request merged**